orchestrator/cli/generator/generator/migration.py

```diff
@@ -45,26 +84,69 @@ def process() -> Generator:
     return dict(process())
 
 
+def update_subscription_model_registry(
+    environment: Environment,
+    config: Dict,
+    product_variants: List[Tuple[Any, Any]],
+    writer: Callable,
+) -> None:
+    template = environment.get_template("subscription_model_registry.j2")
+    product_variable = config.get("variable", "")
+    product_type = config.get("type", "")
+    content = template.render(
+        product=config, product_variants=product_variants, product_types_module=product_types_module
+    )
+
+    path = settings.FOLDER_PREFIX / settings.PRODUCT_REGISTRY_PATH
+    with open(path, "r") as fp:
+        if "SUBSCRIPTION_MODEL_REGISTRY" not in fp.read():
+            fp.close()
+            writer(path, "from orchestrator.domain import SUBSCRIPTION_MODEL_REGISTRY\n\n", append=True)
+    with open(path, "r") as fp:
+        if f"from {product_types_module}.{product_variable} import {product_type}" not in fp.read():
+            fp.close()
+            writer(path, content, append=True)
+        else:
+            logger.warning("not re-updating subscription model registry", product=product_type)
+
+
+def get_revision_info(migration_file: Path) -> Dict:
+    try:
+        with open(migration_file) as stream:
+            original_content = stream.readlines()
+    except FileNotFoundError:
+        logger.error("Migration file not found", path=str(migration_file))
+        return {}
+    else:
+        return extract_revision_info(original_content)
+
+
 def generate_product_migration(context: dict) -> None:
     config = context["config"]
     environment = context["environment"]
     writer = context["writer"]
 
-    if migration_file := create_migration_file(config):
-        path = f"migrations/versions/general/{migration_file}"
-        try:
-            with open(path) as stream:
-                original_content = stream.readlines()
-        except FileNotFoundError:
-            logger.error("Migration file not found", path=path)
-        else:
-            revision_info = extract_revision_info(original_content)
-            variable = get_variable(config)
+    heads = get_heads()
+    if "data" not in heads:
+        create_data_head(context=context, depends_on=heads["schema"])
 
+    if migration_file := create_migration_file(message=f"add {config['name']}", head="data"):
+        if revision_info := get_revision_info(migration_file):
+            if "fixed_inputs" in config and config["fixed_inputs"]:
+                fixed_input_values = [
+                    [(fi["name"], str(value)) for value in fi["values"]] for fi in config["fixed_inputs"]
+                ]
+                fixed_input_combinations = list(itertools.product(*fixed_input_values))
+                product_variants = [
+                    (" ".join([config["name"]] + [field[1] for field in combination]), combination)
```

Comment:
Maybe just extract calculating this value as a small function, to avoid looking at nested for loops.